### ISSUE TITLE
[ILM] Allow option to disable readonly action on index

### ISSUE THREAD

**Issue by kobybr** [elastic/elasticsearch#43426]
While testing ILM, I have encountered instances where logstash persistent queue begins to get backlogged due to ""reason"=>"blocked by: [FORBIDDEN/8/index write (api)]" errors.   This is a result of an index being marked readonly when using the 'force merge' option during the 'warm' phase.   While this normally works fine, there are certain instances that cause this to be problematic.
i.e.
We have a cluster which is used to collect syslog data from hundreds of linux, windows, and networking devices.   These indexes are maintained in **daily** indexes and is based on @timestamp.   The logstash input plugins are beats, tcp, and udp.   Occasionally, logstash will receive logs from previous days (usually windows devices) and try to index into the previous days indexes, which have now been marked readonly.   I'm assuming this old log data is coming from systems that had been shutdown for maintenance, etc...   None the less, the logstash persistent queue begins to backlog since logstash can no longer write to the readonly.   

I understand the reason of not writing to an index during a force merge, but would I believe having the ability to revert the index back to read/write after it is completed would be beneficial.

**Comment by gwbrown** [comment]
I would be hesitant to do this because it's not just writing to an index while the force merge is in progress that can have negative effects, writing to an index after it's been force merged can cause the same problems with very large segments that will never (or almost never) be considered for an automatic merge.

Is adjusting your ILM policy to extend the `min_age` of the `warm` phase a possibility, so that the index wouldn't enter the warm phase until enough time has passed to be confident that all documents have been received? Or do you need to have those indices move off of the hot nodes as soon as possible?

**Comment by kobybr** [comment]
Trying predict a safe `min_age` of the `warm` phase would be difficult to the point of just not wanting to use the forcemerge option.  How can you predict when a filebeat or winlogbeat service is restarted after it failed or was stopped, especially when you are ingesting logs from 1000+ servers/devices.   The force merge api doesn't set an index to readonly.  The ILM process should adhere to the same expectations.   At least give the end-user the ability to disable this added feature.  I would rather have very large segments than have my logstash queue fill to capacity, forcing it stop accepting traffic and thus losing data from udp sources, etc...

**Comment by joegallo** [comment]
Some additional context on #53289, which is a duplicate of this issue.

**Comment by leandrojmp** [comment]
Just had this issue recently, I have a `force merge` action enabled on the `warm` phase on some of my lifecycle policies and it set the index to read-only.

From this [comment](https://github.com/elastic/elasticsearch/issues/53289#issuecomment-597115671) on a duplicated similar issue I understand that this behaviour should be changed, but this is till happening on 7.14.1.

Is this change still in the roadmap?

**Comment by dakrone** [comment]
> Is this change still in the roadmap?

It is, we just have not prioritized it and no one is currently working on it. We still agree with the change (to remove the automatic read-only marking) however.
